static/app/components/events/interfaces/spans/spanBar.tsx

```diff
@@ -216,6 +217,22 @@ class SpanBar extends React.Component<SpanBarProps, SpanBarState> {
     if (this.spanRowDOMRef.current) {
       this.connectObservers();
     }
+    this.scrollIntoView();
+  }
+
+  componentDidUpdate(prevProps: SpanBarProps) {
```

Comment:
@Zylphrex This lifecycle might be better placed in `<SpanTree/>` so that this code block runs at most once, instead of for each span bar.